src/auth.rs

```diff
@@ -53,28 +49,25 @@ pub struct Authorization {
 /// request authentication, and for authenticating outgoing client requests.
 #[derive(Clone, Debug, PartialEq)]
 pub enum AuthData {
```

Comment:
did the old types zeroize on drop, and have we lost that here?

(regardless, I think we should do it)